### ISSUE TITLE
Mining/BP system issues

### ISSUE THREAD

**Issue by VReaperV** [Unvanquished/gameplay#60]
1) A base can be rebuilt over and over again, needlessly prolonging matches. This is mostly the case for aliens because humans can't outrun them for long and it's harder/more time-consuming for them to set up a base. This would be solved by having a sudden death (which is already the case on @cu-kai server).
2) Leeches don't need to be mantained as much as drills. Alien bases mostly consist of egg spam, which doesn't require BP to function. One may argue that the issue in this case is that alien defensive buildings are bad. However, regardless of that humans *have* to use a bunch of defensive buildings. Aliens can have a constant wave of defending aliens with just om and a bunch of eggs, since there's no cooldown on spawn and eggs have enough time to cycle their cooldowns. Leeches are also much easier to hide and don't produce a sound when you're nearby, unlike drills.
3) Armory and medi cannot be rebuilt without BP. Those structures being dead can very easily lead to the humans losing. Aliens, on the other hand, only need the overmind to evolve, and can always regenerate. This is hugely disadvantageous to humans.
4) Armory and medi can't be used with negative BP or if there's not enough BP and they're too far from the reactor. Again, there's no such problem for aliens.
Booster also cannot be rebuilt or function with no BP, but the same being true for armory and medi has a much higher negative impact.

**Comment by Viech** [comment]
Quick response to the points:

1. **We do have a gradual/soft sudden death implemented.** Currently and for many years, build points in the recovery queue recover at an exponentially decreasing rate. The low/negative BP you are experiencing is precisely preventing endless rebuilding. Thus your point as stated is not quite correct.
2. / 3. I agree with the following: Humans require more structures to be functional and this is somewhat unfortunate. I'd like to point out though that while this is _related_ to the mining/build point system, it is not _caused_ by it. Pretty much any way you distribute/allocate/limit buildable resources will suffer from the fact that aliens just need their main building and two eggs to win the game while humans need two more crucial structures and defenses on top of that.
3. See (2).
4. I find this a very good point, except there is again a misunderstanding of how things work (which in itself is understandable, as the way power works is not made transparent to the player). As far as I recall, **buildings power down independently of the build point queue if there are more buildable than the maximum current pool size.** I have made it such that egg and telenode never power down and it would certainly be possible to extend this to other buildables. (At least for the armoury I find this a good idea!)

Side remarks:

- As said on IRC, no objection from me on adding classic sudden death alongside the current system if it is optional and defaults to off.
- Assuming that one allows the armoury to work without power as discussed in (4) and by @sweet235, one could additionally discuss the idea of making the first armoury free. Then, it can always be built and will always function, akin to the overmind.

**Comment by VReaperV** [comment]
> We do have a gradual/soft sudden death implemented. Currently and for many years, build points in the recovery queue recover at an exponentially decreasing rate. The low/negative BP you are experiencing is precisely preventing endless rebuilding.

That is true, but with the current recovery rate values it still allows continuously rebuilding overmind + eggs even in longer matches.

> Pretty much any way you distribute/allocate/limit buildable resources will suffer from the fact that aliens just need their main building and two eggs to win the game while humans need two more crucial structures and defenses on top of that.

I agree, my point is more that you can spam a bunch of leeches around the map, build lots of eggs, and then if the enemy team destroys your leeches it won't have much of an effect on your base. Whereas building a bunch of drills around the map won't be as successful since the enemy team destroying them will shut down your base. In essence, it makes drills less useful than leeches since they will mostly be constrained to your base.

**Comment by cu-kai** [comment]
@Viech
> I n essence, it makes drills less useful than leeches since they will mostly be constrained to your base.

I agree with this, and after some testing on Bunker with SD, @VReaperV fed back to me about making drills re-buildable in SD mode 1, which seemed a sensible thing to do.

Losing a drill hurts humans far more than losing leeches does hurts aliens. As has been said before, aliens can function with just an overmind and eggs. Humans *need* at least an armoury as well as the reactor and telenodes.

**Comment by Viech** [comment]
Most of the issues related to drills ad leeches supporting a minimal base, including problems (3) and (4), should be resolved by an implementation of #61 as both teams would always have the initial build points at their disposal. A question is thus: Would this be sufficient or is alien/human base dependence still something we'd like to balance assuming humans can always build and use an armoury and a medistation? And if so, should we reduce alien capabilities or increase humans, or both?
